src/scheduler/fifo.c

```diff
@@ -1365,6 +1365,7 @@ run_update_resresv(status *policy, int pbs_sd, server_info *sinfo,
 	timed_event *te;			/* used to create timed events */
 	resource_resv *rr;
 	char *err_txt = NULL;
+	char old_state = 0;
```

Comment:
It's not an address so I can not assign NULL to it. The function expects the address of job_state variable and that is why there is a NULL check.